test/libsolidity/syntaxTests/uint_bytes_conversions/literal_legal_lengthening.sol

```diff
@@ -0,0 +1,7 @@
+contract C {
+    bytes2 a = 0xAB;
```

Comment:
Do we have a semantics tests here? Does this do left-padding or right-padding? If it does left-padding, I would rather disallow it.

Reply:
Good question. That is interesting, as the hex literal does convert implicitly, but decimal literal issues a warning.

Reply:
Here's what the compiler says the result is
`..ab............................................................`
(from ABI_CHECK macro on a temp test)